### DISCUSSION TITLE
Interface 'Matchers<R, T>' incorrectly extends interface 'TestingLibraryMatchers<(str: string) => any,            R>'

### DISCUSSION THREAD

**Discussion by shashibhushanvirajaji** [storybookjs/storybook#17995] [Help]
we have installed storybook 6.4 and started to interactive testing on our reach applications,  with latest versions of addon-interactions, jest and testing-library.  
. When we try build the application we are getting below message ...

[ ERROR ]  TypeScript: ./node_modules/@types/jest/index.d.ts:690:15
           Interface 'Matchers<R, T>' incorrectly extends interface 'TestingLibraryMatchers<(str: string) => any,
           R>'.The types returned by 'toHaveAttribute(...)' are incompatible between these types.Type 'void' is not
           assignable to type 'R'.'R' could be instantiated with an arbitrary type which could be unrelated to 'void'.

    L689:  // should be R extends void|Promise<void> but getting dtslint error
    L690:  interface Matchers<R, T = {}> {
    L691:      /**

Can any one share your thoughts on this please . 

I have included all kinds of depdencies like , "@storybook/client-logger": "^6.4.22","@storybook/instrumenter": "^6.4.22", "@storybook/expect": "^27.5.2-0", "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/user-event": "^14.1.1",

**Comment by mrchief** [comment]
I can confirm this doesn't happen on storybook 6.3

**Reply by mrchief** [reply]
Actually, this issues happens with older storybooks also if you remove the lockfile and reinstall. It looks like storybooks' dependencies are pulling in latest `6.5.3` even with older versions 
![image](https://user-images.githubusercontent.com/781818/169446810-a6287031-a4ae-4809-8a95-c3b1921c74da.png)
